modules/launcher/src/main/java/org/mule/runtime/module/launcher/service/ReflectionServiceProviderResolutionHelper.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.module.launcher.service;
+
+import static org.reflections.ReflectionUtils.getAllFields;
+import static org.reflections.ReflectionUtils.withAnnotation;
+import org.mule.runtime.api.service.Service;
+import org.mule.runtime.api.service.ServiceDefinition;
+import org.mule.runtime.api.service.ServiceProvider;
+
+import java.lang.reflect.Field;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+/**
+ * Implements {@link ServiceProviderResolutionHelper} using reflection.
+ */
+public class ReflectionServiceProviderResolutionHelper implements ServiceProviderResolutionHelper
```

Comment:
We have this functionality in several places, MuleContext, Core extensions and now here. Should we consider creating a small context for registration of instances to do dependency injection?. Probably not as part of this commit.